packages/teuchos/comm/src/Teuchos_DefaultMpiComm.hpp

```diff
@@ -261,6 +266,15 @@ class MpiCommRequestBase : public CommRequest<OrdinalType> {
     }
   }
 
+  /// \brief Start the communication request.
+  void start () {
```

Comment:
This method should be virtual, and it should be added to the base class `CommRequest`.  We could use `std::function` and lambda capture to implement `start` for `SerialComm`.  Compare to how Tpetra implements `Tpetra::Details::iallreduce` when `MPI_VERSION < 3`.